src/cats/protocols.cljc

```diff
@@ -97,3 +97,9 @@
 (defprotocol MonadTrans
   "A monad transformer abstraction."
   (-lift [m mv] "Lift a value from the parameterized monad to the transformer."))
+
+;; More info: http://en.wikipedia.org/wiki/Semilattice
+
+(defprotocol JoinSemiLattice
+  "A algebraic structure with element joins."
+  (-join [_ other] "Converge two data structures."))
```

Comment:
Note that there is `cats.core/join` which is used for removing one monadic layer, so we may want to call the join on semilattices differently.